addons/web/static/src/legacy/js/fields/abstract_field_owl.js

```diff
@@ -73,33 +70,14 @@ odoo.define('web.AbstractFieldOwl', function (require) {
             useListener('keydown', this._onKeydown);
             useListener('navigation-move', this._onNavigationMove);
             useEffect(() => this._applyDecorations());
-        }
-        /**
-         * Hack: studio tries to find the field with a selector base on its
-         * name, before it is mounted into the DOM. Ideally, this should be
-         * done in the onMounted hook, but in this case we are too late, and
-         * Studio finds nothing. As a consequence, the field can't be edited
-         * by clicking on its label (or on the row formed by the pair label-field).
-         *
-         * TODO: move this to mounted at some point?
-         *
-         * @override
-         */
-        __patch() {
-            const res = super.__patch(...arguments);
-            this.el.setAttribute('name', this.name);
-            this.el.classList.add('o_field_widget');
-            this.el.classList.toggle('o_quick_editable', this._canQuickEdit);
-            return res;
-        }
-        /**
-         * @async
-         * @param {Object} [nextProps]
-         * @returns {Promise}
-         */
-        async willUpdateProps(nextProps) {
-            this._lastSetValue = undefined;
-            return super.willUpdateProps(nextProps);
+            useEffect(() => {
+                this.el.setAttribute('name', this.name);
+                this.el.classList.add('o_field_widget');
+                this.el.classList.toggle('o_quick_editable', this._canQuickEdit);
+            });
```

Comment:
there was a comment telling that this couldn't be done in mounted, because studio needed it before mounted is called. I manually tested studio and editing views seems to work well. Also, all tests pass.